src/diffusers/pipelines/marigold/pipeline_marigold_depth.py

```diff
@@ -0,0 +1,784 @@
+
+                for t in self.scheduler.timesteps:
+                    batch_latent = torch.cat([batch_image_latent, batch_pred_latent], dim=1)  # [B,8,h,w]
+                    noise = self.unet(batch_latent, t, encoder_hidden_states=batch_text_embedding).sample  # [B,4,h,w]
+                    batch_pred_latent = self.scheduler.step(
+                        noise, t, batch_pred_latent, generator=generator
+                    ).prev_sample  # [B,4,h,w]
+                    progress_bar.update(B)
+
+                clean_latent.append(batch_pred_latent)
+
+                del batch_image_latent, batch_pred_latent, batch_text_embedding, batch_latent, noise
+
+            pred_latent = torch.cat(clean_latent, dim=0)  # [N*E,4,h,w]
+
+            del clean_latent
+
+        del image_latent, batch_empty_text_embedding
+
+        # 6. Decode prediction from latent into pixel space. Model invocation: self.vae.decoder
+        prediction = torch.cat(
+            [
+                self.decode_prediction(pred_latent[i : i + batch_size])
+                for i in range(0, pred_latent.shape[0], batch_size)
+            ],
+            dim=0,
+        )  # [N*E,1,PPH,PPW]
+
+        if not output_latent:
+            pred_latent = None
+
+        # 7. Postprocess predictions
+        prediction = unpad_image(prediction, padding)  # [N*E,1,PH,PW]
+
+        uncertainty = None
+        if ensemble_size > 1:
+            prediction = prediction.reshape(num_images, ensemble_size, *prediction.shape[1:])  # [N,E,1,PH,PW]
+            prediction = [
+                ensemble_depth(prediction[i], output_uncertainty, **(ensembling_kwargs or {}))
+                for i in range(num_images)
+            ]  # [ [[1,1,PH,PW], [1,1,PH,PW]], ... ]
+            prediction, uncertainty = zip(*prediction)  # [[1,1,PH,PW], ... ], [[1,1,PH,PW], ... ]
+            prediction = torch.cat(prediction, dim=0)  # [N,1,PH,PW]
+            uncertainty = torch.cat(uncertainty, dim=0)  # [N,1,PH,PW]
+
+        if match_input_resolution:
+            prediction = resize_antialias(
+                prediction, original_resolution, resample_method_output, is_aa=False
+            )  # [1,1,H,W]
+            if uncertainty is not None and output_uncertainty:
+                uncertainty = resize_antialias(
+                    uncertainty, original_resolution, resample_method_output, is_aa=False
+                )  # [1,1,H,W]
+
+        visualization = None
+        if output_visualization:
+            visualization = [
+                visualize_depth(prediction[i].squeeze(0), **(output_visualization_kwargs or {}))
+                for i in range(num_images)
+            ]  # [PIL.Image, ...]
+
+        if output_prediction_format != "pt":
+            prediction = prediction.cpu().numpy()
+            if uncertainty is not None and output_uncertainty:
+                uncertainty = uncertainty.cpu().numpy()
+            if output_prediction_format == "pil":
```

Comment:
this should be part of the image_processor too

Reply:
Please suggest if this is still necessary given the tricky situation with postprocessing

Reply:
ok :)

Reply:
Resolving this for now so that I can see other unresolved items as todo items; feel free to unresolve this one later - I'm open to discussing this